data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink;
+
+import java.util.Collection;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.services.s3.S3Client;
+
+/**
+ * Class responsible for taking an {@link S3SinkConfig} and creating all the necessary
+ * {@link S3Client} and event accumulate operations.
+ */
+public class S3SinkService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(S3SinkService.class);
+    private final S3SinkConfig s3SinkConfig;
+    private static final int EVENT_QUEUE_SIZE = 100000;
+    private BlockingQueue<Event> eventQueue;

```

Comment:
Why are use using a `BlockingQueue` here?

Reply:
These changes will be addressed as incremental changes, and they will be available in a separate PR.